AutomatedTesting/Gem/PythonTests/atom_renderer/atom_utils/benchmark_utils.py

```diff
@@ -0,0 +1,83 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+"""
+import azlmbr.atom
+import azlmbr.legacy.general as general
+
+FOLDER_PATH = '@user@/Scripts/PerformanceBenchmarks'
+
+class BenchmarkHelper(object):
+    """
+    A helper to capture benchmark data.
+    """
+    def __init__(self, benchmark_name):
+        super().__init__()
+        self.benchmark_name = benchmark_name
+        self.output_path = f'{FOLDER_PATH}/{benchmark_name}'
+        self.done = False
+        self.capturedData = False
+        self.max_frames_to_wait = 200
+
+    def capture_benchmark_metadata(self):
+        """
+        Capture benchmark metadata and block further execution until it has been written to the disk.
+        """
+        self.handler = azlmbr.atom.ProfilingCaptureNotificationBusHandler()
+        self.handler.connect()
+        self.handler.add_callback('OnCaptureBenchmarkMetadataFinished', self.on_data_captured)
+
+        self.done = False
+        self.capturedData = False
+        success = azlmbr.atom.ProfilingCaptureRequestBus(
+            azlmbr.bus.Broadcast, "CaptureBenchmarkMetadata", self.benchmark_name, f'{self.output_path}/benchmark_metadata.json'
+        )
+        if success:
+            self.wait_until_data()
+            general.log('Benchmark metadata captured.')
+        else:
+            general.log('Failed to capture benchmark metadata.')
+        return self.capturedData
+
+    def capture_pass_timestamp(self, frame_number):
+        """
+        Capture benchmark metadata and block further execution until it has been written to the disk.
+        """
+        self.handler = azlmbr.atom.ProfilingCaptureNotificationBusHandler()
+        self.handler.connect()
+        self.handler.add_callback('OnCaptureQueryTimestampFinished', self.on_data_captured)
+
+        self.done = False
+        self.capturedData = False
+        success = azlmbr.atom.ProfilingCaptureRequestBus(
+            azlmbr.bus.Broadcast, "CapturePassTimestamp", f'{self.output_path}/frame{frame_number}_timestamps.json')
+        if success:
+            self.wait_until_data()
```

Comment:
Out of curiosity, how many frames do we usually have to wait?

Reply:
We wait 600 frames total (6 for each capture)